raft/src/main/java/org/apache/kafka/raft/LeaderState.java

```diff
@@ -67,6 +75,39 @@ protected LeaderState(
         }
         this.grantingVoters.addAll(grantingVoters);
         this.log = logContext.logger(LeaderState.class);
+        this.accumulator = Objects.requireNonNull(accumulator, "accumulator must be non-null");
+    }
+
+    public BatchAccumulator<T> accumulator() {
+        return this.accumulator;
+    }
+
+    private static List<Voter> convertToVoters(Set<Integer> voterIds) {
+        return voterIds.stream()
+            .map(follower -> new Voter().setVoterId(follower))
+            .collect(Collectors.toList());
+    }
+
+    public void appendLeaderChangeMessage(int epoch, long baseOffset, long currentTimeMs) {
```

Comment:
We can use the `epoch` field. We can also get rid of `baseOffset` if we move construction of `MemoryRecords` into `BatchAccumulator` as suggested in the other comment.